app/optscale_api/users_api.py

```diff
@@ -24,21 +25,36 @@ async def create_user(
         password: str,
         admin_api_key: str,
         verified: bool = False,
+        check_user_email: bool = False,
     ) -> dict[str, str] | Exception:
         """
         Creates a new user in the system.
 
+        :param check_user_email: If True, we need to check if an invitation has been
+        generated for the given user before creating it.
         :param verified: if verified is True, an access token will be
         generated for the registered user. Otherwise, the token will be None
         :param admin_api_key: the secret admin API key
         :param email: The email of the user.
         :param display_name: The display name of the user
         :param password: The password of the user.
         :return:dict[str, str] : User information.
-        :raises OptScaleAPIResponseError if any error occurs
+        :raises APIResponseError if any error occurs
         contacting the OptScale APIs
         """
-
+        email_check = False
+        if check_user_email:
+            # We need to verify if an invitation has been sent for the given user
+            # If an invitation exists, the user will be created
+            email_check = await validate_user_invitation(email=email)
```

Comment:
As I commented above, it's better IMO to no use an api client within another api client (invitation within user). This will improve readability.